src/main/java/org/ibp/api/java/impl/middleware/germplasm/GermplasmServiceImpl.java

```diff
@@ -313,15 +327,26 @@ private void populateGermplasmPedigree(final List<GermplasmDTO> germplasmDTOList
 	}
 
 	@Override
-	public List<AttributeDTO> getAttributesByGid(
-		final String gid, final List<String> attributeDbIds, final Integer pageSize, final Integer pageNumber) {
-		this.validateGidAndAttributes(gid, attributeDbIds);
-		return this.germplasmDataManager.getAttributesByGid(gid, attributeDbIds, pageSize, pageNumber);
+	public List<AttributeDTO> getAttributesByGUID(
+			final String germplasmUUID, final List<String> attributeDbIds, final Integer pageSize, final Integer pageNumber) {
+		this.validateGuidAndAttributes(germplasmUUID, attributeDbIds);
+		final Optional<GermplasmDTO> germplasmDTO = this.germplasmService.getGermplasmDTOByGUID(germplasmUUID);
+		if (germplasmDTO.isPresent()) {
+			return this.germplasmService.getAttributesByGid(germplasmDTO.get().getGid(), attributeDbIds, pageSize, pageNumber);
+		} else {
+			throw new ApiRuntimeException("Invalid GermplasmDbId");
+		}
+
 	}
 
 	@Override
-	public long countAttributesByGid(final String gid, final List<String> attributeDbIds) {
-		return this.germplasmDataManager.countAttributesByGid(gid, attributeDbIds);
+	public long countAttributesByGUID(final String germplasmUUID, final List<String> attributeDbIds) {
+		final Optional<GermplasmDTO> germplasm = this.germplasmService.getGermplasmDTOByGUID(germplasmUUID);
```

Comment:
1. Call this.germplasmValidator.validateGermplasmGUID and throw the exception right away. You can create a `validateGUID` method:
```
this.errors = new MapBindingResult(new HashMap<String, String>(), Integer.class.getName());
		this.germplasmValidator.validateGermplasmGUID(errors, germplasmDbId);
		if (this.errors.hasErrors()) {
			throw new ResourceNotFoundException(this.errors.getAllErrors().get(0));
		}
```
2. Change MW to `countAttributesByGUID` so you don't need to retrieve by GUID just to get GID
3. Remove throwing of ApiRuntimeExeption("invalid germplasmDbId)